tests/e2e/main.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+
+import sys
+import argparse
+from botocore.exceptions import ClientError
+import boto3
+import time
+import requests
+
+# Create AWS CloudWatch client
+aws_client = boto3.client('logs')
+
+def generate_log_event(log_group_name,log_stream_name):
+    """
+    Upload a log event to the specified AWS CloudWatch Logs log group
+    """
+
+    try:
+        log_event = aws_client.put_log_events(
+            logGroupName  = log_group_name,
+            logStreamName = log_stream_name,
+            logEvents=[
+                {
+                    'timestamp': epoch_timestamp_in_ms,
+                    'message': message_content
+                },
+            ],
+        )
+        print("Generated log event: %s" % log_event)
+    except ClientError as e:
+        if e.response['Error']['Code'] == 'DataAlreadyAcceptedException' or e.response['Error']['Code'] == 'InvalidSequenceTokenException':
+            print("Given log batch has already been accepeted. Retrying with the following sequence token: " + e.response['expectedSequenceToken'])
+            log_event_retry = aws_client.put_log_events(
```

Comment:
this is semi e2e test because you own generate log, better solution will be to generate log by some service like Lambda ora ECS or something else

Reply:
We discussed this and we are perfectly fine with this solution. External service brings very little difference for what we want to test so this is mostly theoretical thing. Our solution starts at cloudwatch log group and ends in cluster API.

Reply:
As Marek mentioned, MVP was to generate a log without a specific source just to test the forwarding solution.